### PR TITLE
[ADD] payment_fix_register_token: Fix account.payment.register wizard with 'payment'…

### PR DESCRIPTION
**Pull request by smetl**
… module

The payment token is no longer available when registering a payment for invoices.
This new module aims to restore this lost feature during the payment-pocalyse in stable version.
Of course, this module must be merged in existing modules during the forwart-port in master.

task: 2352628

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

**Comment by smetl** [comment]
@qdp-odoo I would like to have your opinion on this. The payment token flow on `account.payment` seems weird because handled directly in `action_post`. I think it should be better to remove the override of `action_post` in `account.payment`, then doesn't allow anymore a user to `create` an `account.payment` directly with a payment token and then, add a hook in `account.payment.register` to manage the `payment.transaction`/`payment.token` properly. Also, we could find a way to prevent the creation of a cancelled payment when the transaction failed and improve the feedback returned to the user in such case.

@AdrienHorgnies cc

File: addons/payment_fix/models/account_payment_register.py [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+        compute='_compute_payment_token_id',
+        domain='''[
+            (payment_method_code == 'electronic', '=', 1),
+            ('company_id', '=', company_id),
+            ('acquirer_id.capture_manually', '=', False),
+            ('acquirer_id.journal_id', '=', journal_id),
+            ('partner_id', 'in', suitable_payment_token_partner_ids),
+        ]''',
+        help="Note that tokens from acquirers set to only authorize transactions (instead of capturing the amount) are "
+             "not available.")
```
Comment (qdp-odoo) [inline addons/payment_fix/models/account_payment_register.py:23]: heu? ok? why not the capturing _manually ones? was it like that before?

**Review by qdp-odoo — CHANGES_REQUESTED** [review]
missing .pot file for the new module

File: addons/payment_fix/models/account_payment_register.py [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+    _inherit = 'account.payment.register'
+
+    # == Business fields ==
+    payment_token_id = fields.Many2one(
+        comodel_name='payment.token',
+        string="Saved payment token",
+        store=True, readonly=False,
+        compute='_compute_payment_token_id',
+        domain='''[
+            (payment_method_code == 'electronic', '=', 1),
```
[the remark this answers is not in the document]
Comment (smetl) [reply addons/payment_fix/models/account_payment_register.py:16]: Probably a magic trick only known by XMO. It seems to work. Copy pasted from:
https://github.com/odoo/odoo/blob/14.0/addons/payment/models/account_payment.py#L21

File: addons/payment_fix/models/account_payment_register.py (hunk 1 above)
Comment (smetl) [reply addons/payment_fix/models/account_payment_register.py:23]: Copy pasted from:
https://github.com/odoo/odoo/blob/14.0/addons/payment/models/account_payment.py#L22

**Comment by smetl** [comment]
@qdp-odoo `.pot` file added

**Comment by KangOl** [comment]
This new module is really badly named. What about `payment_fix_register_token` or something more specific to what it actually fix.

**Comment by smetl** [comment]
@KangOl why not :)

**Review by qdp-odoo — CHANGES_REQUESTED** [review]
looks ok, but there's that little thing

File: addons/account/tests/test_account_move_out_invoice.py [hunk 3]
```diff
@@ -1319,6 +1319,23 @@ def test_out_invoice_line_onchange_analytic(self):
...
+            'code': 'TEST1'
+        })
+
+        self.invoice.write({'invoice_line_ids': [(1, self.invoice.invoice_line_ids.ids[0], {
+            'analytic_account_id': analytic_account.id,
+        })]})
+
+        self.assertRecordValues(self.invoice.invoice_line_ids, [
+            {'analytic_account_id': analytic_account.id},
+            {'analytic_account_id': False},
```
[the remark this answers is not in the document]
Comment (smetl) [reply addons/account/tests/test_account_move_out_invoice.py]: Indeed. It comes from another branch... my bad

**Comment by qdp-odoo** [comment]
@robodoo r+

**Comment by qdp-odoo** [comment]
@robodoo rebase-ff

**Pull request closed without merging**